setup.py

```diff
@@ -66,22 +67,26 @@ def _maintain_symlinks(symlink_type, base_path):
     except (IOError, OSError) as e:
         # IOError on py2, OSError on py3.  Both have errno
         if e.errno == 2:
-            # SYMLINKS_CACHE doesn't exist.  Fallback to trying to create the
-            # cache now.  Will work if we're running directly from a git
-            # checkout or from an sdist created earlier.
-            symlink_data = {'script': _find_symlinks('bin'),
-                            'library': _find_symlinks('lib', '.py'),
-                            }
-
-            # Sanity check that something we know should be a symlink was
-            # found.  We'll take that to mean that the current directory
-            # structure properly reflects symlinks in the git repo
-            if 'ansible-playbook' in symlink_data['script']['ansible']:
-                _cache_symlinks(symlink_data)
-            else:
-                raise
+            missing_symlink_cache = True
         else:
             raise
+
+    if missing_symlink_cache:
```

Comment:
This doesn't need to be extracted out of try/except. You could've just changed only `raise` statement.